larpandora/LArPandoraInterface/LArPandoraGeometry.cxx

```diff
@@ -29,57 +29,76 @@ void LArPandoraGeometry::LoadDetectorGaps(LArDetectorGapList &listOfGaps)
     LArDriftVolumeList driftVolumeList;
     LArPandoraGeometry::LoadGeometry(driftVolumeList);
 
+    // ATTN: Expectations here are that the input geometry corresponds to either a single or dual phase LArTPC.
+    art::ServiceHandle<geo::Geometry const> theGeometry;
+    const bool isDualPhase(theGeometry->MaxPlanes() == 2);
```

Comment:
Maybe the optimal way would be an xml parameter to ensure there is zero ambiguity? I guess right now it could impact because in the end there is assumptions under the hood when it comes to computing gaps (especially gaps only in Y/Z directions). Also as John mentioned, I tried to keep the same definition of the isDualPhase boolean variable in everything I added (maybe I missed it and there is another definition somewhere in the code ?)

Reply:
Consistency is good!
This topic, that I think all three of us have asked about, is probably better aimed at LArSoft/DUNE, rather than us trying to figure it out here.  

Actually, it might be worth querying what the signal type for each view/plane is:
https://nusoft.fnal.gov/larsoft/doxsvn/html/classgeo_1_1GeometryCore.html#a748e324d604cec19be4553eaa4a8e208
I'd hope that both planes return a collection-type signal for dual phase, which can never ever be the case for single phase...